### DISCUSSION TITLE
How to Set Roles and Groups - Login via Self-Hosted Keycloak

### DISCUSSION THREAD

**Discussion by iamsid** [outline/outline#4829] [Self Hosting] [answered]
Hi.
We have installed Outline on a DO droplet via CapRover. In the same droplet we also have a self-hosted Keycloak.
Our Outline instance is working well and users can log in via Keycloak.
But, I don't see any roles against the members, nor any groups - both of which have been defined in Keycloak's client configuration.

I tried searching in issues and discussions but could not find anything relevant. Please help me understand how to get the roles/groups to work?

Outline: v0.66.3

**Comment by tommoor** [comment, marked as answer]
There is no concept of mapping roles or groups from OIDC providers to those inside Outline at this time.

**Reply by iamsid** [reply]
So how do I set them in Outline? I don't see any option to manage these from inside Outline either.
Am I missing some configuration?

**Reply by tommoor** [reply]
Group management is available in the settings sidebar as an admin, if you don't see it then your account doesn't have an admin role.

<img width="311" alt="image" src="https://user-images.githubusercontent.com/380914/216984664-1115b670-37c4-4e58-ab56-8a4ac1810722.png">

**Reply by iamsid** [reply]
I do see these options, but no option to set role or create group. How do I define myself as admin? I just logged in with the Keycloak credentials. No other changes done in Outline.
![image](https://user-images.githubusercontent.com/17705415/217007612-a62e87bb-5193-40e5-b9e0-9413fc15c68b.png)

**Reply by iamsid** [reply]
@tommoor Any ideas on what I am doing incorrectly?

**Reply by tommoor** [reply]
The first user created is an admin, your user is not an admin. You can either sign-in with the first created user or just change yourself to an admin in the database directly

**Reply by iamsid** [reply]
Not sure what went wrong, but I was the first user. I could see two users in the DB, first me, and second the other test account.
Both were not admin. Setting myself as admin worked.
Thanks. :)

**Reply by Nico-L** [reply]
Hello,

Same issue here, self-hosting + keycloak. First user is not admin. I had to get to the database to change the isAdmin column.
